### DISCUSSION TITLE
Repeat mode

### DISCUSSION THREAD

**Discussion by SecretArea** [Syncplay/syncplay#559] [Archive]
Do you think it would be a good idea to add a replay function at the end of a track? Currently, playback is paused if the video clip ends.
Syncplay ignores VLC's "loop" auto-repeat.

**Comment by yuroyami** [comment]
I thought of this too. Perchance, a few people would make use of this feature, maybe playing any kind of audio like Music. If thats implemented, I will implement it too in [Syncplay Android](www.github.com/Reddnek/syncplay-android).

That can be implemented in a lot of alternative ways to avoid media player incompatibility with this feature. For instance, tracking the playback position and when it reaches the end the player seeks back to 0:00 and resumes. Maybe a little pause would be a nice touch, as an option.


This may be included in the room's settings, like chat availability.

**Comment by Et0h** [comment]
While not ideal, the official Syncplay client already has two options to loop at the end of the playlist, they just require you to manually change the loopAtEndOfPlaylist and if relevant loopSingleFiles values in the client_settings group of the .syncplay/syncplay.ini file.

Ideally everyone on a session should have the same settings.

This is not documented within Syncplay itself, but it is mentioned at https://syncplay.pl/guide/client/
> ### **Configuration options not accessible through GUI**
> There are a number of settings which can be changed by modifying the syncplay.ini / .syncplay file. If your change breaks something then you can just delete the entry to restore it to the default, but you might still want to make a backup of your configuration file before making changes. You can figure out what these do by searching for the setting name in the Syncplay source code.
> 
> **[GUI] options relating to mpv OSD and chat input:**
> * **Numeric:** chatMaxLines, chatTopMargin, chatLeftMargin, chatBottomMargin, chatOSDMargin, notificationTimeout, alertTimeout, chatTimeout
> * **Boolean (True/False):** chatMoveOSD
> 
> **[client_settings] options relating to playback syncing and playlists, file changing:**
> * **Numeric:** slowdownThreshold, rewindThreshold, fastforwardThreshold
> * **Boolean (True/False):** loopAtEndOfPlaylist, loopSingleFiles, onlySwitchToTrustedDomains, autoplayRequireSameFilenames

**Comment by Et0h** [comment]
This discussion is being locked as the feature exists and is documented in the Client guide for anyone who is interested in it. There are no plans to add this feature to the GUI as we want the interface to be as streamlined as possible.
